src/pudl/workspace/setup.py

```diff
@@ -4,310 +4,120 @@
 import pathlib
 import shutil
 from pathlib import Path
-from typing import IO
 
-import yaml
-from dotenv import load_dotenv
+from pydantic import BaseSettings
 
 import pudl.logging_helpers
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-def get_defaults(
+def set_path_overrides(
     input_dir: str | None = None,
     output_dir: str | None = None,
-    yaml_file: IO | None = None,
-    default_pudl_yaml: Path | None = Path.home() / ".pudl.yml",
-) -> dict[str, str]:
-    """Derive PUDL workspace paths from specified input/output directories.
+) -> None:
+    """Set PUDL_INPUT and/or PUDL_OUTPUT env variables.
 
-    Determines input/output directory locations from YAML, then overrides with
-    env vars, then overrides with keywords passed in.
-
-    Input/output workspace roots can be the same directories.
+    Args:
+        input_dir: if set, overrides PUDL_INPUT env variable.
+        output_dir: if set, overrides PUDL_OUTPUT env variable.
+    """
+    if input_dir:
```

Comment:
The reason we want to use env vars here instead of just initializing the `PudlPaths()` with different parameters is so that child processes get the right settings automatically, right?

Reply:
Yup, that's the point. We call `set_path_overrides` early on and then assume that `PudlPaths` always returns the right things from then on. I think we may have passed pudl_settings, but this was not actually hermetic and many call-sites still got their own instance via `get_defaults()` which would break the consistency.